### PR TITLE
Make emplace infer class ctor safety

### PR DESCRIPTION
**Pull request by edi33416**

### CODE REVIEW

**Comment by andralex** [comment]
@JackStouffer actually the preexisting code is kept, only one new overload got added.

**Comment by JackStouffer** [comment]
Sorry, I didn't see the extra overload.

**Comment by andralex** [comment]
@edi33416 looks like the tester is stuck, could you please push to restart?

File: std/conv.d [hunk 1]
```diff
@@ -4517,25 +4508,98 @@ if (is(T == class))
...
+
+    class UnsafeClass
+    {
+        int x;
+        @system this(int x) { this.x = x; }
+    }
+
+    auto buf2 = new void[__traits(classInstanceSize, UnsafeClass)];
+    auto support2 = (() @trusted => cast(UnsafeClass)(buf2.ptr))();
+    static assert(!__traits(compiles, emplace!UnsafeClass(support2, 5)));
```
Comment (edi33416) [inline std/conv.d]: @wilzbach @CyberShadow 
Hello guys!

I'm giving you this ping as I believe we have a bug in the style checker.
The files that get generated remove the `@safe` from the unittests which cause these static asserts to fail.

Could you please have a look at this? Is this behaviour by design? What is your input on this?

Cheers

**Comment by edi33416** [comment]
The win32 tests are failing with the following error
`Error: more than 32767 symbols in object file`

What should I make of this?
Why isn't a separate object generated for each `.d` file?

**Comment by edi33416** [comment]
Ok, so I changed the `win32.mak` makefile to split the `unittest6.obj` into two objects  and tested on a Windows 32bit VM and the tests pass.

This brings me to the following:
- the 32767 max symbols, I'm guessing, is a dmd limitation
- why are we bundling multiple `.d` source files into a single object instead of compiling each file to it's corresponding `.obj`?

**Comment by ZombineDev** [comment]
@edi33416 now that #5779 has been merged, can you please rebase?

**Pull request merged**